component/componentstatus/status.go

```diff
@@ -1,12 +1,20 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-package component // import "go.opentelemetry.io/collector/component"
+package componentstatus // import "go.opentelemetry.io/collector/component/componentstatus"
 
 import (
 	"time"
 )
 
+type StatusReporter interface {
```

Comment:
Our guidelines state:

```
Constant values of an enumeration should be prefixed with the enumeration type name in the name
```

So `componentstatus .StatusStarting` and `componentstatus .StatusOk` are correct since the type is named `Status`.  Should we rename the type to something else?